cloud-data-protection/src/components/demo/demo.tsx

```diff
@@ -0,0 +1,190 @@
+            .then(() => setSelectedFile(undefined))
+            .catch((e) => onError(e))
+            .finally(() => stopLoading());
+    }
+
+    const onFileSelect = (e: any) => {
+        const file: File = e.target.files[0];
+
+        if (file.size > DemoService.maxFileSize) {
+            enqueueSnackbar('The selected file is too big. The upload limit is 25MB.', snackbarOptions);
+
+            setSelectedFile(undefined);
+
+            e.target.value = null;
+
+            return;
+        }
+
+        setSelectedFile(e.target.files[0]);
+    }
+
+    const onFileIdChange = async () => {
+        cancelTokenSource = http.CancelToken.source();
+
+        startLoading();
+
+        await demoService.getFileInfo(fileId, cancelTokenSource.token)
+            .then((result) => setFileInfo(result))
+            .catch((e) => console.log(e))
+            .finally(() => stopLoading());
+    }
+
+    const copyToClipboard = async (e: any) => {
+        await navigator.clipboard.writeText(e.target.innerText);
+
+        enqueueSnackbar('Code has been copied to the clipboard', { ...snackbarOptions, autoHideDuration: 2500 });
+    }
+
+    const download = async (decrypt: boolean) => {
+        if (!fileId) {
+            enqueueSnackbar('Please enter a file code.');
+            return;
+        }
+
+        cancelTokenSource = http.CancelToken.source();
+
+        startLoading();
+
+        await demoService.downloadFile(fileId, decrypt, cancelTokenSource.token)
+            .catch((e) => onError(e))
+            .finally(() => stopLoading());
+    }
+
+    const onError = (e: any) => {
+        if (e instanceof String) {
+            enqueueSnackbar(e, snackbarOptions);
+        }
+    }
+
+    return (
+        <div className='backup-demo'>
+            <Typography variant='h1'>Backup demo</Typography>
+            <div className='backup-demo__upload'>
+                <Typography variant='h2'>Upload file</Typography>
+                <p>
+                    Get a taste of the performance and security Cloud Data Protection offers. Upload a file to get started.
+                </p>
+                <form  onSubmit={(e) => onSubmit(e)}>
+                    <Button className='backup-demo__upload__form__select-file' variant='contained' component="label">
+                        {selectedFile ?
+                            <span>{selectedFile.name} ({formatBytes(selectedFile.size)})</span> :
+                            <span>Select a file</span>
+                        }
+                        <input type="file" hidden onChange={(e) => onFileSelect(e)} />
+                    </Button>
+
+                    {uploadedFile &&
+                        <div className='backup-demo__uploaded-file'>
+                            Your file has been uploaded. You can access it later by saving this code. Click it to copy to the clipboard: <code className='backup-demo__uploaded-file__id' onClick={e => copyToClipboard(e)}>{uploadedFile.storageId}</code>
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/olivierbouchomsfontys/cloud-data-protection/pullRequest?prid=7431242)
```suggestion
                            Your file has been uploaded. You can access it later by saving this code. Click it to copy to the clipboard: <code className='backup-demo__uploaded-file__id' onClick={(e) => copyToClipboard(e)}>{uploadedFile.storageId}</code>
```